src/cmakeProject.ts

```diff
@@ -1650,7 +1650,7 @@ export class CMakeProject {
                         } else {
                             buildLogger.info(localize('build.finished.with.code', 'Build finished with exit code {0}', rc));
                         }
-                        const fileDiags = consumer!.compileConsumer.resolveDiagnostics(drv!.binaryDir);
+                        const fileDiags: FileDiagnostic[] | undefined = drv!.config.parseBuildDiagnostics ? consumer!.compileConsumer.resolveDiagnostics(drv!.binaryDir) : undefined;
```

Comment:
Can you set it to `[]` instead of undefined?  I think we should clear the list of diagnostics if the user changes the setting after getting some diagnostics listed in the Problems.